tests/test_transpilers_blocks.py

```diff
@@ -0,0 +1,180 @@
+import pytest
+from qibo import Circuit, gates
+
+from qibolab.transpilers.blocks import (
+    Block,
+    BlockingError,
+    _find_previous_gates,
+    _find_successive_gates,
+    block_decomposition,
+    count_2q_gates,
+    count_multi_qubit_gates,
+    gates_on_qubit,
+    initial_block_decomposition,
+    remove_gates,
+)
+
+
+def test_count_2q_gates():
+    block = Block(qubits=(0, 1), gates=[gates.CZ(0, 1), gates.CZ(0, 1), gates.H(0)])
+    assert block.count_2q_gates() == 2
+
+
+def test_rename():
+    block = Block(qubits=(0, 1), gates=[gates.CZ(0, 1)])
+    block.rename("renamed_block")
+    assert block.name == "renamed_block"
+
+
+def test_add_gate_and_entanglement():
+    block = Block(qubits=(0, 1), gates=[gates.H(0)])
+    assert block.entangled == False
+    block.add_gate(gates.CZ(0, 1))
+    assert block.entangled == True
+    assert block.count_2q_gates() == 1
+
+
+def test_add_gate_error():
+    block = Block(qubits=(0, 1), gates=[gates.CZ(0, 1)])
+    with pytest.raises(BlockingError):
+        block.add_gate(gates.CZ(0, 2))
+
+
+def test_fuse_blocks():
+    block_1 = Block(qubits=(0, 1), gates=[gates.CZ(0, 1)])
+    block_2 = Block(qubits=(0, 1), gates=[gates.H(0)])
+    fused = block_1.fuse(block_2)
+    assert fused.qubits == (0, 1)
+    assert fused.entangled == True
+    assert fused.count_2q_gates() == 1
```

Comment:
being a little pedantic, it might be more general to check that the fused block consists of the gates coming from the two original blocks